src/server/req_modify.c

```diff
@@ -848,90 +911,124 @@ req_modifyReservation(struct batch_request *preq)
-		}
-
 		psatl = (svrattrl *)GET_NEXT(psatl->al_link);
 	}
 
 
 	if (presv->ri_wattr[RESV_ATR_state].at_val.at_long == RESV_RUNNING && num_jobs) {
-		if ((presv->ri_alter_flags & RESV_DURATION_MODIFIED) && (presv->ri_alter_flags & RESV_END_TIME_MODIFIED)) {
-			resv_revert_alter_times(presv);
+		if ((presv->ri_alter.ra_flags & RESV_DURATION_MODIFIED) && (presv->ri_alter.ra_flags & RESV_END_TIME_MODIFIED)) {
+			resv_revert_alter(presv);
 			req_reject(PBSE_RESV_NOT_EMPTY, 0, preq);
 			return;
 		}
 	}
-	resc_access_perm = resc_access_perm_save; /* restore perm */
+
+
+	bad = 0;
+	psatl = (svrattrl *)GET_NEXT(preq->rq_ind.rq_modify.rq_attr);
+	if (psatl) {
+		rc = modify_resv_attr(presv, psatl, preq->rq_perm, &bad);
+		if (rc != 0) {
+			reply_badattr(rc, bad, psatl, preq);
+			resv_revert_alter(presv);
+			return;
+		}
+	}
+	resc_access_perm = resc_access_perm_save;
 
 	new_end_time = presv->ri_wattr[RESV_ATR_start].at_val.at_long + presv->ri_wattr[RESV_ATR_duration].at_val.at_long;
 
-	if ((presv->ri_alter_flags & RESV_DURATION_MODIFIED) && presv->ri_alter_etime == 0) {
+	if ((presv->ri_alter.ra_flags & RESV_DURATION_MODIFIED) && presv->ri_alter.ra_etime == 0) {
 		if (!is_standing || new_end_time < next_occr_start) {
-			presv->ri_alter_etime = presv->ri_wattr[RESV_ATR_end].at_val.at_long;
+			presv->ri_alter.ra_etime = presv->ri_wattr[RESV_ATR_end].at_val.at_long;
 		} else {
 			log_event(PBSEVENT_RESV, PBS_EVENTCLASS_RESV, LOG_INFO,
-				preq->rq_ind.rq_modify.rq_objname, msg_stdg_resv_occr_conflict);
+				  preq->rq_ind.rq_modify.rq_objname, msg_stdg_resv_occr_conflict);
 			req_reject(PBSE_STDG_RESV_OCCR_CONFLICT, 0, preq);
-			resv_revert_alter_times(presv);
+			resv_revert_alter(presv);
 			return;
 		}
 	}
 
-	if ((presv->ri_alter_flags & RESV_DURATION_MODIFIED) && presv->ri_alter_stime == 0) {
+	if ((presv->ri_alter.ra_flags & RESV_DURATION_MODIFIED) && presv->ri_alter.ra_stime == 0) {
 		if (!is_standing || new_end_time < next_occr_start) {
-			presv->ri_alter_stime = presv->ri_wattr[RESV_ATR_start].at_val.at_long;
+			presv->ri_alter.ra_stime = presv->ri_wattr[RESV_ATR_start].at_val.at_long;
 		} else {
 			log_event(PBSEVENT_RESV, PBS_EVENTCLASS_RESV, LOG_INFO,
-				preq->rq_ind.rq_modify.rq_objname, msg_stdg_resv_occr_conflict);
+				  preq->rq_ind.rq_modify.rq_objname, msg_stdg_resv_occr_conflict);
 			req_reject(PBSE_STDG_RESV_OCCR_CONFLICT, 0, preq);
-			resv_revert_alter_times(presv);
+			resv_revert_alter(presv);
 			return;
 		}
 	}
 
+	if (presv->ri_alter.ra_flags & RESV_SELECT_MODIFIED) {
+		presc = find_resc_entry(&presv->ri_wattr[RESV_ATR_resource], prdef);
+		make_schedselect(&presv->ri_wattr[(int) RESV_ATR_resource], presc, NULL, &presv->ri_wattr[(int) RESV_ATR_SchedSelect]);
+		if (is_select_smaller(presv->ri_wattr[RESV_ATR_SchedSelect_orig].at_val.at_str, presv->ri_wattr[RESV_ATR_SchedSelect].at_val.at_str) == 0) {
+			req_reject(PBSE_SELECT_NOT_SUBSET, 0, preq);
+			resv_revert_alter(presv);
+			return;
+		}
+	}
```

Comment:
You should do this check before we set all the attributes of the ralter, to be consistent with the other options.

Reply:
The call to is_select_smaller() is made on schedselect to schedselect.  I need to make the schedselect first before I can compare it.

Reply:
Can't you make the sched select in the switch-case statement?
It seems this is a test we can do before setting all the attributes, instead of comparing the new with original schedselects, you can compare the resv schedselect and a schedselect created from the attribute from the ralter request.

Reply:
I can't.  The schedselect is made from the select, so the select attribute has to be set.  The attributes are only set after the select.